### DISCUSSION TITLE
Window resize

### DISCUSSION THREAD

**Discussion by valleymouth** [vsg-dev/VulkanSceneGraph#589] [Q&A] [answered]
Hi guys,
Let's say I have a main window and another small one (like a PiP). When I resize the main window, the small one also resizes, this apparently is the default behaviour in VSG. How would one go about to maintain the size of the small one unaltered?
Cheers,
Denis

**Comment by robertosfield** [comment]
On Thu, 17 Nov 2022 at 14:38, valleymouth ***@***.***> wrote:

> Let's say I have a main window and another small one (like a PiP). When I
> resize the main window, the small one also resizes, this apparently is the
> default behaviour in VSG. How would one go about to maintain the size of
> the small one unaltered?
>
Just tried the vsgwindows example and the two windows resize independently
as they should:

    $ vsgwindows models/lz.vsgt models/teapot.vsgt

There is no code in the VSG to synchronize the resizing of two windows
together, so rather than default behavior I don't even know how it might be
possible.

Could you provide an example that illustrates the problem you are seeing?

Thanks,
Robert.

**Comment by valleymouth** [comment]
Hi Robert,
Sorry, my explanation was horrible. Remember the multiple camera example you developed for us at Arrival? They have multiple cameras by having multiple RenderGraphs inside a CommandGraph. If you resize the window all the RenderGraphs get resized.
I think the place it is done is in WindowResizeHandler maybe?
Cheers,
Denis

**Comment by robertosfield** [comment, marked as answer]
HI Dennis.

On Fri, 18 Nov 2022 at 12:10, valleymouth ***@***.***> wrote:

> Sorry, my explanation was horrible. Remember the multiple camera example
> you developed for us at Arrival? They have multiple cameras by having
> multiple RenderGraphs inside a CommandGraph. If you resize the window all
> the RenderGraphs get resized.
> I think the place it is done is in WindowResizeHandler maybe?Message ID:
> ***@***.***
> com>
>

Ahh that's a different situation to multiple windows.

Multiple views on the same window will all be resized together.  If you
want different behavior then you'll need to create your own
WindowResizeHandler and assign that to the RenderGraph.  I haven't ever
tried this though.... so I don't know if there are any gotchas.

The vsgmultiviews example might be a good place to add a test for a custom
WindowResizeHandler.

Cheers,
Robert.

**Reply by valleymouth** [reply]
Hi Robert,
OK, I will have a go at it. Thank you so much for always being available to help! :)
Cheers,
Denis

**Reply by redflare2023** [reply]
hi,Dennis,
I have the same problem, Can you give me some advice?

cheers,
David

**Comment by redflare2023** [comment]
hi,Robert,
I have the same problem. the RenderGraph class already has one WindowResizeHandler member val. The problem is how can i add another one to the RenderGraph? the next is how is  the WindowResizeHandler dealt?

cheers,
David

**Comment by robertosfield** [comment]
Could you please create a new thread for your questions as Github
Discussion and a clear explanation of what you are trying to achieve and
what you have attempted to do so far.  Thanks.

**Reply by redflare2023** [reply]
> Could you please create a new thread for your questions as Github Discussion and a clear explanation of what you are trying to achieve and what you have attempted to do so far. Thanks.

Ok，I will create a new thread. thank for your response!
